packages/cojson/src/coValueState.ts

```diff
@@ -1,53 +1,50 @@
+import { PeerState } from "./PeerState.js";
 import { CoValueCore } from "./coValueCore.js";
+import { RawCoID } from "./ids.js";
 import { PeerID } from "./sync.js";
 
-function createResolvablePromise<T>() {
-  let resolve!: (value: T) => void;
-
-  const promise = new Promise<T>((res) => {
-    resolve = res;
-  });
+export const CO_VALUE_LOADING_MAX_RETRIES = 5;
 
-  return { promise, resolve };
+export class CoValueUnknownState {
+  type = "unknown" as const;
 }
 
-class CoValueUnknownState {
-  type = "unknown" as const;
-  private peers: Map<
+export class CoValueLoadingState {
+  type = "loading" as const;
+  private peers = new Map<
     PeerID,
-    ReturnType<typeof createResolvablePromise<"available" | "unavailable">>
-  >;
-  private resolve: (value: "available" | "unavailable") => void;
+    ReturnType<typeof createResolvablePromise<void>>
+  >();
+  private resolve: (value: CoValueCore | "unavailable") => void;
 
-  ready: Promise<"available" | "unavailable">;
+  result: Promise<CoValueCore | "unavailable">;
+  resolution: undefined | CoValueCore | "unavailable";
```

Comment:
Added this `resolution` attribute to be able to read the resolved state syncronously